core/src/main/java/org/infinispan/persistence/sifs/Index.java

```diff
@@ -199,11 +203,12 @@ private boolean checkForExistingIndexSizeFile() {
                sizePerSegment.set(i, value);
             }
 
-            if (indexCount.read() != -1) {
-               log.tracef("Previous index file has more bytes than desired, assuming index is a different format");
-            } else {
-               validCount = true;
+            try {
+               maxSeqId = UnsignedNumeric.readUnsignedLong(indexCount);
+            } catch (EOFException e) {
+               log.tracef("Index didn't contain sequence id, will need to calculate, still retaining index");
             }
+            validCount = true;
```

Comment:
Seems `validCount` is always true now.

Reply:
Not always, if an IOException is thrown instead of an EOF then it is false.